lib/facter/vault_version.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+# Fact: vault_version
+#
+# Purpose: Retrieve vault version if installed
+#
+Facter.add(:vault_version) do
+  setcode do
+    if Facter::Util::Resolution.which('vault')
```

Comment:
I think the cleaner solution is to confine the fact 
```ruby
confine { Facter::Util::Resolution.which('ip') }
setcode do
 ...
```

Reply:
Sorry for late reply. You are right, I changed the code accordingly.